client_apis/python/src/tests/integration/settings_test.py

```diff
@@ -0,0 +1,275 @@
+            cb.communication_settings_modify(mail_server_type="badtype")
+
+        self.assertEqual(err.exception.response.status_code, 400)
+        return
+
+    def test_communication_settings_mail_server_own_invalid(self):
+        settings = cb.communication_settings()
+        self.assertIsNotNone(settings)
+        self.assertIn("mail_server_type", settings)
+
+        mail_settings = {
+            "mail_server_type": "own",
+            "smtp_connection_type": "ssl",
+            "smtp_server": None,
+            "smtp_port": None,
+            "smtp_username": None,
+            "smtp_password": None
+        }
+
+        with self.assertRaises(requests.HTTPError) as err:
+            cb.communication_settings_modify(settings=mail_settings)
+
+        self.assertEqual(err.exception.response.status_code, 400)
+
+        mail_settings = {
+            "mail_server_type": "own",
+            "smtp_connection_type": "bad_type",
+            "smtp_server": "localhost",
+            "smtp_port": "25",
+            "smtp_username": "username",
+            "smtp_password": "password"
+        }
+
+        with self.assertRaises(requests.HTTPError) as err:
+            cb.communication_settings_modify(settings=mail_settings)
+
+        self.assertEqual(err.exception.response.status_code, 400)
+        return
+
+    def test_concurrent_license_info(self):
+        end = datetime.now()
+        start = end - timedelta(days=30)
+        result = cb.concurrent_license_info(start, end)
+        self.assertIsNotNone(result)
+        return
+
+    def test_info(self):
+        result = cb.info()
+        self.assertIsNotNone(result)
+        return
+
+    def test_check_for_new_feeds(self):
+        result = cb.check_for_new_feeds()
+        self.assertIsNotNone(result)
+        self.assertIn("result", result)
+        self.assertEqual(result["result"], "success")
+        return
+
+    def test_get_platform_server_config(self):
+        config = cb.get_platform_server_config()
+        self.assertIsNotNone(config)
+        return
+
+    def test_get_platform_server_config_insufficient_permissions(self):
+        user = TestDataGen.gen_user()
+        cb.user_add_from_data(
+            username=user["username"],
+            password=user["password"],
+            confirm_password=user["password"],
+            first_name=user["first_name"],
+            last_name=user["last_name"],
+            teams=user["teams"],
+            email=user["email"],
+            global_admin=False
+        )
+
+        auth_token = Creds4Token.get_token(url, user["username"], user["password"])
+        self.assertIsNotNone(auth_token)
+
+        non_ga_cb = CbApi(server=url, ssl_verify=False, token=auth_token, client_validation_enabled=False)
+
+        with self.assertRaises(requests.HTTPError) as err:
+            non_ga_cb.get_platform_server_config()
+
+        self.assertEqual(err.exception.response.status_code, 405)
```

Comment:
I'm not sure that expecting an HTTP 405 is the proper thing here.  I think 405s are supposed to be used when the route doesn't support the HTTP action (e.g. a GET in this case).  I would think it more appropriate to expect a 403.